core/gui/src/app/workspace/service/report-generation/report-generation.service.ts

```diff
@@ -67,27 +69,244 @@ export class ReportGenerationService {
   }
 
   /**
-   * Generates an HTML file containing the workflow snapshot and triggers a download of the file.
-   * @param {string} workflowSnapshotURL - The base64-encoded PNG image URL of the workflow snapshot.
-   * @param {string} workflowName - The name of the workflow.
+   * Retrieves and processes results for all specified operators within the workflow.
+   * This function iterates over each operator ID, fetches the corresponding result details via `retrieveOperatorInfoReport`,
+   * and collects these results into an array. The function returns an observable that emits the processed results,
+   * which can be used to generate a comprehensive HTML report or for further processing.
+   *
+   * @param {string[]} operatorIds - An array of operator IDs representing each operator in the workflow.
+   *
+   * @returns {Observable<{operatorId: string, html: string}[]>} - An observable that emits an array of objects,
+   * each containing an `operatorId` and its corresponding HTML representation of the result.
+   * This result array can be used to generate an HTML report or for other purposes.
+   */
+
+  public getAllOperatorResults(operatorIds: string[]): Observable<{ operatorId: string; html: string }[]> {
+    return new Observable(observer => {
+      const allResults: { operatorId: string; html: string }[] = [];
+
+      const promises = operatorIds.map(operatorId => {
+        return this.retrieveOperatorInfoReport(operatorId, allResults);
+      });
+
+      Promise.all(promises)
+        .then(() => {
+          observer.next(allResults);
+          observer.complete();
+        })
+        .catch(error => {
+          observer.error("Error in retrieving operator results: " + error.message);
+        });
+    });
+  }
+
+  /**
+   * Retrieves and processes detailed results for a specified operator, generating a structured HTML representation.
+   * This function covers multiple cases, including handling paginated results, snapshot data, and scenarios where
+   * no results are found for the operator. The resulting HTML content is stored in the `allResults` array.
+   *
+   * @param {string} operatorId - The unique identifier of the operator for which results are being processed.
+   * @param {any} operatorInfo - Metadata and configuration details of the operator, utilized to render a JSON editor within the HTML.
+   * @param {Array<{operatorId: string, html: string}>} allResults - An array used to collect HTML content for each operator's processed result.
+   *
+   * @returns {Promise<void>} - A promise that resolves once the operator's results have been processed and the HTML has been appended to `allResults`.
    */
-  public generateReportAsHtml(workflowSnapshotURL: string, workflowName: string): void {
+
+  public retrieveOperatorInfoReport(
```

Comment:
make this also return a Observe<{operatorId: string; operatorResultHtml: string}[]>, and remove `allResults` from the parameter list.